src/client/Client.cc

```diff
@@ -2348,6 +2348,15 @@ void Client::handle_client_session(const MConstRef<MClientSession>& m)
 	mount_cond.notify_all();
       else
 	connect_mds_targets(from);
+
+      if (cct->_conf.get_val<bool>("client_reconnect_stale") && !last_unsafe_reqs.empty()) {
+        ldout(cct, 10) << "replay unsafe request before do anything" << dendl;
+        while (!last_unsafe_reqs.empty()) {
+          session->unsafe_requests.push_back(&last_unsafe_reqs.front()->unsafe_item);
+        }
```

Comment:
Why send all the unsafe requests through this session ? The `last_unsafe_reqs` list you added should also could be used by other sessions, right ?

Reply:
all of these unsafe requests added into `last_unsafe_reqs` when session killed by mds, used by new session reconnect to mds for keep the consistency

Reply:
> Why send all the unsafe requests through this session ? The `last_unsafe_reqs` list you added should also could be used by other sessions, right ?

all of these unsafe requests added into `last_unsafe_reqs` when session killed by mds, used by new session reconnect to mds for keep the consistency

Reply:
I mean you are adding all the requests, which maybe from some other sessions, into the global `last_unsafe_reqs` list. But here you are trying to send all the unsafe requests through a single session ?

Reply:
yes, cause all of these unsafe requests are collected from mds rebooted, I think it's ok to process it by the first mds reconnect. Or there is something wrong in multiple mds situation? I'm not deep in multiple mds.

Reply:
Not multiple MDS case, but for the multiple clients case.

Reply:
> Not multiple MDS case, but for the multiple clients case.

Sorry, this is the client side code, it should be the multiple mds case.

Reply:
don't mind, so do we still have any problems here?

Reply:
You still need to send the unsafe requests to the dedicated MDS instead of sending all the unsafe requests to a single MDS. We have fixed several multiple MDS related bugs for the client requests. Such as some requests were suppose to be sent to the auth MDS instead of the replica ones and to fix some bugs we must do that.

Reply:
ok, I'll change it later

Reply:
Changed, now these unsafe_requests will be added to specific xlist<MetaRequest*> differentiated by req->mds